hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/callback/TestThrowExceptionCallback.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.callback;
+
+import org.apache.hudi.client.BaseHoodieClient;
+import org.apache.hudi.exception.HoodieIOException;
+
+/**
+ * A test {@link HoodieClientInitCallback} implementation to throw an exception.
+ */
+public class TestThrowExceptionCallback implements HoodieClientInitCallback {
```

Comment:
is it possible to keep all these as internal classes rather than individual classes by itself ?

Reply:
I tried inner class but it does not work well with the class loader.  So I rather put them in independent classes.

Reply:
OK, the inner classes have to be static to work.  Now all test implementation are in inner classes.